### PR TITLE
feat: adds support for composite prices

### PR DESCRIPTION
**Pull request by ankitmhn**

### CODE REVIEW

File: clients/pricing-client/src/openapi.d.ts [hunk 1]
```diff
@@ -1051,7 +1059,14 @@ declare namespace Components {
             /**
              * Contains price item configurations, per price component, when the main price item is a [price bundle](/api/pricing#tag/dynamic_price_schema).
              */
-            item_components?: PriceItemComponent[];
+            item_components?: /**
+             * Represents a price item
+             * example:
+             * {
+             *   "$ref": "#/components/examples/order-with-simple-prices"
```
Comment (paulofph) [inline clients/pricing-client/src/openapi.d.ts:1066]: I think this example is not correct. It's an `order` not a `price`

File: clients/pricing-client/src/openapi.d.ts [hunk 2]
```diff
@@ -1123,14 +1149,182 @@ declare namespace Components {
...
+            id?: string;
+            metadata?: /* A set of key-value pairs used to store meta data information about an entity. */ MetaData;
+            /**
+             * Contains price item configurations, per price component, when the main price item is a [price bundle](/api/pricing#tag/dynamic_price_schema).
+             */
+            item_components?: /**
+             * Represents a price item
+             * example:
+             * {
+             *   "$ref": "#/components/examples/order-with-simple-prices"
```
Comment (paulofph) [inline clients/pricing-client/src/openapi.d.ts:1252]: Same here 🤔

File: clients/pricing-client/src/openapi.d.ts [hunk 3]
```diff
@@ -353,6 +353,240 @@ declare namespace Components {
...
+         */
+        export interface CompositePrice {
+            [name: string]: any;
+            /**
+             * Whether the price can be used for new purchases.
+             */
+            active?: boolean;
+            /**
+             * A brief description of the price.
+             */
```
Comment (jpinho) [inline clients/pricing-client/src/openapi.d.ts:371]: We're missing the `currency` field here.

**Review by jpinho — APPROVED** [review]
Please fix the comments and merge 🙇 
Good work 👍

File: clients/pricing-client/src/openapi.d.ts (hunk 3 above)
Comment (paulofph) [reply clients/pricing-client/src/openapi.d.ts:371]: On it.

**Pull request merged**